### PR TITLE
Feature/error messages 

### PR DESCRIPTION
**Pull request by norling**
Minor changes in order to improve error messages.

### CODE REVIEW

**Review by shreyasshivakumara — DISMISSED** [review]
Changes look neat! :)

File: main.go [hunk 1]
```diff
@@ -52,7 +61,8 @@ func main() {
 	go hc.RunHealthChecks()
 
 	if config.Server.cert != "" && config.Server.key != "" {
-		if e := http.ListenAndServeTLS(":8000", config.Server.cert, config.Server.key, nil); e != nil {
+		if e := http.ListenAndServeTLS(":8000", config.Server.cert,
+			config.Server.key, nil); e != nil {
```
Comment (jbygdell) [inline main.go:65]: Why the line break here?
It actually makes the code harder to read.

File: main.go [hunk 2]
```diff
@@ -34,12 +41,14 @@ func main() {
 	// Load keys for JWT verification
 	if config.Server.jwtpubkeyurl != "" {
 		if err := auth.getjwtpubkey(config.Server.jwtpubkeyurl); err != nil {
-			log.Panicf("Error while getting key %s: %v", config.Server.jwtpubkeyurl, err)
+			log.Panicf("Error while getting key %s: %v",
+				config.Server.jwtpubkeyurl, err)
```
Comment (jbygdell) [inline main.go:45]: why the line break?

File: main.go [hunk 3]
```diff
@@ -34,12 +41,14 @@ func main() {
...
 		}
 	}
 	if config.Server.jwtpubkeypath != "" {
 		if err := auth.getjwtkey(config.Server.jwtpubkeypath); err != nil {
-			log.Panicf("Error while getting key %s: %v", config.Server.jwtpubkeypath, err)
+			log.Panicf("Error while getting key %s: %v",
+				config.Server.jwtpubkeypath, err)
```
Comment (jbygdell) [inline main.go:51]: why the line break?

**Review by jbygdell — CHANGES_REQUESTED** [review]

File: dev_utils/README.md [hunk 4]
```diff
@@ -1,20 +1,26 @@
...
+started from this directory using:
 
 ```bash
-docker-compose up -d
+docker compose up
 ```
+(use the `-d` flag if you wish to run docker in the background).
 
 Then it's possible to trace all the requests that come to minio by first
 putting the following in the hosts array your `~/.mc/config.json` file:
```
Comment (aaperis) [inline dev_utils/README.md:18]:
```suggestion
putting the following in the hosts array of your `~/.mc/config.json` file:
```

File: proxy.go [hunk 5]
```diff
@@ -116,6 +116,15 @@ func (p *Proxy) allowedResponse(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Writing non-200 to the response before the headers propagates the error
```
Comment (aaperis) [inline proxy.go:119]:
```suggestion
	// Writing non-200 to the response before the headers propagate the error
```

File: proxy.go [hunk 6]
```diff
@@ -116,6 +116,15 @@ func (p *Proxy) allowedResponse(w http.ResponseWriter, r *http.Request) {
...
+	// to the s3cmd client.
+	// Writing 200 here breaks uploads though, and writing non-200 codes after
+	// the headers result in the error message always being
```
Comment (aaperis) [inline proxy.go:122]:
```suggestion
	// the headers results in the error message always being
```

**Review by aaperis — APPROVED** [review]
A few minor typos. I also agree with not breaking lines too often. 
Otherwise it looks good :-)

File: main.go (hunk 3 above)
Comment (norling) [reply main.go:51]: just a habit to limit lines to 80 characters! I'll revert it :)

**Review by jbygdell — APPROVED** [review]

**Review by dbampalikis — APPROVED** [review]

**Review by kostas-kou — APPROVED** [review]

**Pull request merged**